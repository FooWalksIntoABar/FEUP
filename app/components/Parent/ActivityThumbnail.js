import React from 'react'
import { View, Image, StyleSheet } from 'react-native'
import { PropTypes } from 'prop-types'

export class SortableListItemThumbnail extends React.Component {
  styles = StyleSheet.create({
    view: {
      height: 50,
      width: 50,
      borderRadius: 25,
      marginRight: 15,
      justifyContent: 'center',
      alignItems: 'center'
    },
    image: {
      width: '60%',
      height: '60%'
    }
  })

  render () {
    return (
      <View style={[this.styles.view, { backgroundColor: this.props.color }]}>
        <Image source={this.props.source} style={this.styles.image} resizeMode={'contain'} />
      </View>
    )
  }
}

SortableListItemThumbnail.propTypes = {
  color: PropTypes.string.isRequired,
  source: PropTypes.number.isRequired
}
