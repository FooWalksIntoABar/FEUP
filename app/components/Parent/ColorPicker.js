import React, { Component } from 'react'
import { Picker, Item } from 'native-base'
import PropTypes from 'prop-types'

export class ColorPicker extends Component {
  onValueChange = value => {
    this.props.onColorChange(this.props.colors[value].code)
  }

  render () {
    const pickerItems = this.props.colors.map((color, index) => (<Picker.Item label={color.name} value={index} key={index} />))
    return (
      <Item picker style={{ borderColor: this.props.color }}>
        <Picker
          mode='dropdown'
          style={{ width: 120 }}
          selectedValue={this.props.colors.findIndex(color => color.code === this.props.color)}
          onValueChange={this.onValueChange}
        >
          {pickerItems}
        </Picker>
      </Item>
    )
  }
}

ColorPicker.propTypes = {
  color: PropTypes.string.isRequired,
  colors: PropTypes.array.isRequired,
  onColorChange: PropTypes.func.isRequired
}
