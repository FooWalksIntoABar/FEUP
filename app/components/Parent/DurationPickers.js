import React, { Component } from 'react'
import { View } from 'react-native'
import { Item, Label, Input, Button, Text } from 'native-base'
import PropTypes from 'prop-types'

export class DurationPickers extends Component {
  constructor (props) {
    super(props)
    this.state = {
      detailedView: false
    }
  }

  toggleDetailedView = () => {
    this.setState({ detailedView: !this.state.detailedView })
  }

  changeGoalValue = (value) => {
    this.props.onDurationChange({ goal: value.replace(/[^\d.-]/g, ''), interpolate: this.state.detailedView })
  }

  changeMaxValue = (value) => {
    this.props.onDurationChange({ max: value.replace(/[^\d.-]/g, ''), interpolate: this.state.detailedView })
  }

  changeMinValue = (value) => {
    this.props.onDurationChange({ min: value.replace(/[^\d.-]/g, ''), interpolate: this.state.detailedView })
  }

  render () {
    return (
      <View>
        <Item stackedLabel>
          <Label>{this.state.detailedView ? 'Duração ideal (minutos)' : 'Duração (minutos)'}</Label>
          <Input keyboardType='numeric' textContentType='telephoneNumber' value={this.props.time.goal} onChangeText={this.changeGoalValue} />
        </Item>
        {this.state.detailedView && <Item stackedLabel>
          <Label>Duração Minima (minutos)</Label>
          <Input keyboardType='numeric' textContentType='telephoneNumber' value={this.props.time.min} onChangeText={this.changeMinValue} />
        </Item>}
        {this.state.detailedView && <Item stackedLabel>
          <Label>Duração Máxima (minutos)</Label>
          <Input keyboardType='numeric' textContentType='telephoneNumber' value={this.props.time.max} onChangeText={this.changeMaxValue} />
        </Item>}
        <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
          <Button transparent onPress={this.toggleDetailedView}>
            <Text style={{ color: this.props.color }}>{this.state.detailedView ? 'Menos opções de duração' : 'Mais opções de duração'}</Text>
          </Button>
        </View>
      </View>
    )
  }
}

DurationPickers.propTypes = {
  onDurationChange: PropTypes.func,
  time: PropTypes.object.isRequired,
  color: PropTypes.string.isRequired
}
