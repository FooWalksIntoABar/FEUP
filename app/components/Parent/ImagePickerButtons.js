import React, { Component } from 'react'
import { ImagePicker, Permissions } from 'expo'
import { Label, Button, Text, Item } from 'native-base'
import { View, StyleSheet, Image } from 'react-native'
import PropTypes from 'prop-types'

import Images, { availableImages } from '../../assets/images/images'
import { ItemPicker } from './ItemPicker'

export class ImagePickerButtons extends Component {
  constructor (props) {
    super(props)
    this.state = {
      photoView: true
    }
  }

  togglePhotoView = () => {
    this.setState({ photoView: !this.state.photoView })
  }

  imagePicker = async (camera) => {
    const options = { allowsEditing: true, aspect: [5, 3] }
    let result

    if (camera) {
      await Permissions.askAsync(Permissions.CAMERA)
      result = await ImagePicker.launchCameraAsync({ ...options, mediaTypes: 'Images' })
    } else {
      await Permissions.askAsync(Permissions.CAMERA_ROLL)
      result = await ImagePicker.launchImageLibraryAsync(options)
    }

    if (!result.cancelled) {
      this.props.onPhotoChange(result.uri)
    }
  }

  onImageChanged = (index) => {
    this.props.onImageChange(availableImages.map(item => item.image)[index])
  }

  getSelectedImageName = () => {
    const image = availableImages.find(item => item.image === this.props.image)
    if (image) return image.name
    else return undefined
  }

  renderOption = () => {
    if (this.state.photoView) {
      return (
        <View style={styles.itemContainer}>
          {this.props.photo && <View style={styles.imageContainer}>
            <Image style={styles.photoPreview} source={this.props.photo} resizeMode={'contain'} />
          </View>}
          <View style={styles.imageButtonContainer}>
            <Button style={[styles.imageButton, { backgroundColor: this.props.color }]} onPress={() => this.imagePicker(true)}>
              <Text>Camara</Text>
            </Button>
            <Button style={[styles.imageButton, { backgroundColor: this.props.color }]} onPress={() => this.imagePicker(false)}>
              <Text>Galeria</Text>
            </Button>
          </View>
        </View>)
    } else {
      return (
        <View style={styles.itemContainer}>
          {this.props.image && <View style={styles.imageContainer}>
            <Image style={styles.imagePreview} source={Images[this.props.image]} resizeMode={'contain'} />
          </View>}
          <Item style={{ borderColor: 'transparent' }}>
            <ItemPicker items={availableImages.map(item => item.name)} selected={this.getSelectedImageName()} onValueChange={this.onImageChanged} />
          </Item>
        </View>
      )
    }
  }

  render () {
    return (
      <View>
        <Item stackedLabel>
          <Label>Imagem</Label>
          {this.renderOption()}
        </Item>
        <View style={{ flexDirection: 'row', justifyContent: 'flex-end' }}>
          <Button transparent onPress={this.togglePhotoView}>
            <Text style={{ color: this.props.color }}>{this.state.photoView ? 'Escolher imagem' : 'Escolher fotografia'}</Text>
          </Button>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  itemContainer: {
    flexDirection: 'column',
    width: '100%'
  },
  imageButtonContainer: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'flex-start'
  },
  imageButton: {
    flex: 1,
    marginRight: 15,
    marginVertical: 15,
    justifyContent: 'center'
  },
  photoPreview: {
    width: '100%',
    aspectRatio: 1.666
  },
  imagePreview: {
    width: 100,
    height: 100
  },
  imageContainer: {
    flexGrow: 1,
    paddingRight: 15,
    marginTop: 15,
    alignItems: 'center'
  }
})

ImagePickerButtons.propTypes = {
  color: PropTypes.string.isRequired,
  onPhotoChange: PropTypes.func.isRequired,
  onImageChange: PropTypes.func.isRequired,
  photo: PropTypes.object,
  image: PropTypes.string
}
