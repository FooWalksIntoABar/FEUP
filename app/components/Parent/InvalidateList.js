import React from 'react'
import { List, ListItem, Left, Text, Right, Icon, Button } from 'native-base'
import { View, Image } from 'react-native'
import { SortableListItemThumbnail } from './SortableListItemThumbnail'
import PropTypes from 'prop-types'
import { getSource } from '../../helpers/GetSource'
import { red } from '../../styles/Colors'
import images from '../../assets/images/images'

export class InvalidateList extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      activities: this.props.activities
    }
    if (this.props.activities == null) {
      this.state.activities = []
    }
  }

  formatTime = time => {
    if (time <= 0) {
      return '0:00'
    }

    let minutes = Math.floor(time / 60)

    let seconds = Math.floor(time % 60)
    if (seconds < 10) seconds = '0' + seconds

    return `${minutes}:${seconds}`
  }

  render () {
    console.log(this.props.activities)
    const activities = this.props.activities.map((activity, index) => (
      <ListItem button onPress={this.onItemPress} key={index}>
        <Left>
          <SortableListItemThumbnail isPhoto={activity.photo !== null} color={activity.color} source={getSource(activity)} />
          <View style={{ flex: 1 }}>
            <Text style={{ alignSelf: 'flex-start' }}>{activity.title}</Text>
            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
              <Image source={images.ui.star} style={[{ height: 12, width: 12, margin: 2 }, activity.reward < 1 && { tintColor: 'lightgrey' }]} />
              <Image source={images.ui.star} style={[{ height: 12, width: 12, margin: 2 }, activity.reward < 2 && { tintColor: 'lightgrey' }]} />
              <Image source={images.ui.star} style={[{ height: 12, width: 12, margin: 2 }, activity.reward < 3 && { tintColor: 'lightgrey' }]} />
              <Text style={{ alignSelf: 'flex-start' }} note numberOfLines={1}>{`- ${activity.reward ? 'Completada' : 'Falhada'} em ${this.formatTime(activity.elapsedTime)}`}</Text>
            </View>
          </View>
        </Left>
        <Right>
          <Button rounded onPress={() => this.props.onInvalidatePress(index)} style={{ backgroundColor: red }}>
            <Icon name={'md-close'} />
          </Button>
        </Right>
      </ListItem>))

    return (
      <List style={{ width: '100%' }}>
        {activities}
      </List>
    )
  }
}

InvalidateList.propTypes = {
  activities: PropTypes.array.isRequired,
  onInvalidatePress: PropTypes.func.isRequired
}
