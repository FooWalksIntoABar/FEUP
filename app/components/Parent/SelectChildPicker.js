import React, { Component } from 'react'
import { Picker } from 'native-base'
import { View, Image, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

export class SelectChildPicker extends Component {
  render () {
    const pickerItems = this.props.children.map((child, index) => (<Picker.Item label={child.name} value={index} key={index} />))

    return (
      <View style={styles.childPickerContainer}>
        <View style={styles.imageView}>
          <Image style={styles.photo} source={{ uri: this.props.children[this.props.selected].image }} />
        </View>
        <Picker
          mode='dropdown'
          style={{ flexGrow: 1, height: '100%' }}
          selectedValue={this.props.selected}
          onValueChange={this.props.onChildChanged}
          itemTextStyle={{ color: 'red' }}
        >
          {pickerItems}
        </Picker>
      </View>
    )
  }
}

SelectChildPicker.propTypes = {
  children: PropTypes.array.isRequired,
  selected: PropTypes.number.isRequired,
  onChildChanged: PropTypes.func.isRequired
}

const styles = StyleSheet.create({
  imageView: {
    height: 70,
    width: 70,
    borderRadius: 35,
    margin: 15,
    justifyContent: 'center',
    alignItems: 'center',
    overflow: 'hidden'
  },
  photo: {
    width: '100%',
    height: '100%'
  },
  childPickerContainer: {
    width: '100%',
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: 'lightgrey',
    alignItems: 'center'
  }
})
