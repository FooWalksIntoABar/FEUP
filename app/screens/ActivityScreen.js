import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { setActivityStatus, nextActivity } from '../actions/gameActions'
import { addStars } from '../actions/childActions'
import { Image, Text, View, StatusBar, BackHandler } from 'react-native'

import { ProgressBar } from '../components/Activity/ProgressBar'
import { ProgressClock } from '../components/Activity/ProgressClock'
import { CompleteButton } from '../components/Activity/CompleteButton'
import { PauseButton } from '../components/Activity/PauseButton'
import { CancelButton } from '../components/Activity/CancelButton'
import { RewardsModal } from '../components/RewardsModal/RewardsModal'
import Images from '../assets/images/images'

import styles from '../styles/Activity.style'

class ActivityScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      elapsedTime: 0,
      progressType: '',
      isPhoto: this.props.activity.photo !== undefined,
      updateRate: 100, // ms
      isPaused: false,
      isCompleted: false,
      isCompletable: false
    }

    this.pauseActivity = this.pauseActivity.bind(this)
    this.cancelActivity = this.cancelActivity.bind(this)
    this.completeActivity = this.completeActivity.bind(this)
    this.resumeActivity = this.resumeActivity.bind(this)
    this.nextActivity = this.nextActivity.bind(this)
    this.backToMenu = this.backToMenu.bind(this)
  }

  componentDidMount () {
    BackHandler.addEventListener('hardwareBackPres', this.cancelActivity)
    this.interval = setInterval(() => {
      if (this.state.isPaused) return
      if (this.state.elapsedTime >= this.props.activity.time.max) this.completeActivity()
      if (this.state.elapsedTime >= this.props.activity.time.min) {
        this.state.isCompletable = true
      }

      this.setState(() => {
        return { elapsedTime: this.state.elapsedTime + this.state.updateRate / 1000 }
      })
    }, this.state.updateRate)
  }

  componentWillUnmount () {
    BackHandler.removeEventListener('hardwareBackPress', this.cancelActivity)
    clearInterval(this.interval)
  }

  pauseActivity () {
    this.setState(() => ({ isPaused: true }))
  }

  cancelActivity () {
    this.props.navigation.popToTop()
    return true // To be called by andrdid back button
  }

  completeActivity () {
    clearInterval(this.interval)
    const status = {
      completed: true,
      reward: this.state.elapsedTime > this.props.activity.time.max ? 0
        : this.state.elapsedTime < this.props.activity.time.goal ? 3
          : this.state.elapsedTime < this.props.activity.time.goal + (this.props.activity.time.max - this.props.activity.time.goal) / 2 ? 2 : 1,
      time: parseInt(this.state.elapsedTime)
    }
    this.props.setActivityStatus(this.props.activity, status)
    this.setState(() => { return { isCompleted: true } })
  }

  nextActivity () {
    if (this.props.activity.status) this.props.addStars(this.props.activity.status.reward)
    this.props.nextActivity()
    this.props.navigation.replace('Activity')
  }

  backToMenu () {
    if (this.props.activity.status) this.props.addStars(this.props.activity.status.reward)
    this.props.navigation.popToTop()
  }

  resumeActivity () {
    this.setState(() => ({ isPaused: false }))
  }

  returnURIorImage = () => {
    // TODO: This checks whether the photo attribute is type URI and should probably just eventually be totally changed to URI.
    if (this.props.activity.photo !== undefined && this.props.activity.photo.includes('file://')) {
      return { uri: this.props.activity.photo }
    } else {
      return Images[this.state.isPhoto ? this.props.activity.photo : this.props.activity.image]
    }
  }

  render () {
    return (
      <View style={[{ backgroundColor: this.props.activity.color }, styles.activityScreen]} >
        <StatusBar hidden />
        <Image
          style={this.state.isPhoto ? styles.photo : styles.image}
          resizeMode={this.state.isPhoto ? 'cover' : 'center'}
          source={this.returnURIorImage()} />
        <View style={styles.titleContainer}>
          <Text style={this.state.isPhoto ? styles.photoTitle : styles.title}>{this.props.activity.title}</Text>
        </View>
        {this.props.progressType === 'clock' && !this.state.isCompleted && <ProgressClock showTimer={this.props.showTimer} elapsedTime={this.state.elapsedTime} activityTimes={this.props.activity.time} isPaused={this.state.isPaused} />}
        {!this.state.isCompleted && <View style={styles.buttonContainer}>
          {this.props.progressType === 'bar' && <ProgressBar showTimer={this.props.showTimer} elapsedTime={this.state.elapsedTime} activityTimes={this.props.activity.time} isPaused={this.state.isPaused} />}
          <CancelButton style={styles.smallButton} cancelActivity={this.cancelActivity} />
          <PauseButton style={styles.smallButton} pauseActivity={this.pauseActivity} resumeActivity={this.resumeActivity} isPaused={this.state.isPaused} />
          <CompleteButton style={styles.largeButton} isCompletable={this.state.isCompletable} completeActivity={this.completeActivity} />
        </View>}
        <RewardsModal
          currentActivity={this.props.currentActivity}
          activities={this.props.activities}
          nextPress={this.nextActivity}
          backPress={this.backToMenu} />
      </View>
    )
  }
}

export default connect(
  state => ({
    progressType: state.settings.activityProgressType,
    showTimer: state.settings.activityShowTimer,
    activity: state.game.routines[state.game.currentRoutine].activities[state.game.currentActivity],
    activities: state.game.routines[state.game.currentRoutine].activities,
    currentActivity: state.game.currentActivity
  }),
  dispatch => ({
    setActivityStatus: (activity, status) => dispatch(setActivityStatus(activity, status)),
    nextActivity: () => dispatch(nextActivity()),
    addStars: stars => dispatch(addStars(stars))
  })
)(ActivityScreen)

ActivityScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  progressType: PropTypes.string.isRequired,
  showTimer: PropTypes.bool.isRequired,
  currentActivity: PropTypes.number.isRequired,
  activities: PropTypes.array.isRequired,
  activity: PropTypes.object.isRequired,
  setActivityStatus: PropTypes.func.isRequired,
  nextActivity: PropTypes.func.isRequired,
  addStars: PropTypes.func.isRequired
}
