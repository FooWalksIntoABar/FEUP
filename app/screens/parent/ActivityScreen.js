import React from 'react'
import { Content, Spinner } from 'native-base'
import { InvalidateList } from '../../components/Parent/InvalidateList'
import { SelectChildPicker } from '../../components/Parent/SelectChildPicker'
import EnvVars from '../../constants/EnviromentVars'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

export class ActivityScreen extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      children: [],
      activities: [],
      selectedChild: 0,
      loading: true
    }

    this.removeStars = this.removeStars.bind(this)
  }

  componentDidMount () {
    this.getChildren()
  }

  getChildren () {
    let url = `${EnvVars.apiUrl}routine_manager/children?userEmail=${this.props.loggedUserEmail}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.setState({ children: JSON.parse(responseJson.response) }, this.getActivities)
        } else {
          console.log('Ta mal')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  getActivities () {
    let url = `${EnvVars.apiUrl}routine_manager/history?id=${this.state.children[this.state.selectedChild].id}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          const activities = responseJson.response === null ? [] : JSON.parse(responseJson.response)
          this.setState({ activities, loading: false })
        } else {
          console.log('Ta mal')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  removeStars = (index) => {
    fetch(EnvVars.apiUrl + 'routine_manager/remove-stars/', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        id: this.state.children[this.state.selectedChild].id,
        reward: this.state.activities[index].reward,
        activityID: this.state.activities[index].id
      })
    }).then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.setState({ activities: this.state.activities.filter((_, i) => i !== index) })
        } else {

        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  onChildChanged = child => {
    this.setState({ selectedChild: child, loading: true }, this.getActivities)
  }

  render () {
    if (this.state.loading) return (<Content contentContainerStyle={{ flexGrow: 1, justifyContent: 'center' }}><Spinner /></Content>)
    else {
      return (
        <Content>
          <SelectChildPicker children={this.state.children} selected={this.state.selectedChild} onChildChanged={this.onChildChanged} />
          <InvalidateList activities={this.state.activities} onInvalidatePress={this.removeStars} />
        </Content>
      )
    }
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    loggedUserEmail: state.user.email
  })
)(ActivityScreen)

ActivityScreen.propTypes = {
  loggedUserEmail: PropTypes.string.isRequired
}
