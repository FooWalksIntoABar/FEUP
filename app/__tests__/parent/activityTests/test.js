import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import '../../../__mock__/xhr-mock'

import { ActivityScreen } from '../../../screens/parent/ActivityScreen'
import { SelectChildPicker } from '../../../components/Parent/SelectChildPicker'
import { InvalidateList } from '../../../components/Parent/InvalidateList'

import { activitiesDone } from '../../../constants/mockTestData'

configure({ adapter: new Adapter() })

describe('ActivityScreen snapshot', () => {
  it('renders ActivityScreen correctly', () => {
    const wrapper = shallow(<ActivityScreen
      loggedUserEmail='email@aleatorio.com' />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  /*
  it('renders ActivityScreen correctly', () => {
    const wrapper = shallow(<ActivityScreen
      loggedUserEmail='email@aleatorio.com' />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setState({ activities: activitiesDone, children: ['Child'] })
    wrapper.find(SelectChildPicker).at(0).props().onChildChanged(0)
    wrapper.find(InvalidateList).at(0).props().onInvalidatePress()
    wrapper.unmount()
  })
  */
})
