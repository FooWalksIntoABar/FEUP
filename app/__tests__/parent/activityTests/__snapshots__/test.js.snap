// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityScreen snapshot renders ActivityScreen correctly 1`] = `
<Styled(Content)>
  <SelectChildPicker
    onChildChanged={[Function]}
    selected={0}
  >
    <Component />
    <Component />
    <Component />
  </SelectChildPicker>
  <InvalidateList
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "elapsedTime": 40,
          "photo": "https://hniesfp.imgix.net/8/images/detailed/14/EA1A7075.jpg?fit=fill&bg=0FFF&w=1500&h=1000&auto=format,compress",
          "reward": 3,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "elapsedTime": 120,
          "image": "toothbrush",
          "reward": 3,
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "elapsedTime": 40,
          "image": "socks",
          "reward": 2,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "elapsedTime": 34,
          "image": "shower",
          "reward": 3,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "elapsedTime": 3000,
          "image": "bag",
          "reward": 0,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "elapsedTime": 20,
          "image": "sneakers",
          "reward": 3,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "elapsedTime": 242,
          "image": "breakfast",
          "reward": 3,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "elapsedTime": 27,
          "image": "comb",
          "reward": 2,
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    onInvalidatePress={[Function]}
  />
</Styled(Content)>
`;
